examples/customLayers/customBilayer.cpp

```diff
@@ -70,49 +65,44 @@ extern "C" {
 
         // Manually deal with hydration for layers in
         // this example.
-        double oxSLD = (oxideHydration * *bulkOut) + ((1 - oxideHydration) * oxideSLD);
-        double headSLD = (headHydration * *bulkOut) + ((1 - headHydration) * SLDhead);
-        double tailSLD = (bilayerHydration * *bulkOut) + ((1 - bilayerHydration) * SLDtail);
+        double oxSLD = (oxideHydration * bulkOut[contrast]) + ((1 - oxideHydration) * oxideSLD);
+        double headSLD = (headHydration * bulkOut[contrast]) + ((1 - headHydration) * SLDhead);
+        double tailSLD = (bilayerHydration * bulkOut[contrast]) + ((1 - bilayerHydration) * SLDtail);
```

Comment:
How does this code work? At a glance multiplying by a pointer and a single element of a vector seem to be completely different operations.

Reply:
In the original code `(bilayerHydration * *bulkOut)` is equivalent to `(bilayerHydration * bulkOut[0])` the former used pointer dereferencing to get the first value in the array. This worked because we previously passed the bulk out for the current contrast only so the first value was always `bulkOut[contrast]` but now we pass the full array